### ISSUE TITLE
AADAdministrativeUnit: Unescaped characters lead to error when exporting configuration to Excel format

### ISSUE THREAD

**Issue by GruberMarkus** [microsoft/Microsoft365DSC#3172]
#### Details of the scenario you tried and the problem that is occurring
1. Export AADAdministrativeUnit.
2. A member of type MSFT_MicrosoftGraphMember has an apostrophy in it's name, for example
  ```
                MSFT_MicrosoftGraphMember { 
                    Type = 'Device'
                    Identity = 'XXX’s iPhone'
                }
   ```
3. Exporting the configuration to an Excel file using New-M365DSCReportFromConfiguration leads to the error message shown in the following section.

#### Verbose logs showing the problem
Sorry, German only at the moment.
The first message is about the line `Identity = 'XXX’s iPhone'`. The remaining lines are follow-up errors.
```
WARNUNG: ConvertTo-DSCObject: "s" (line 335): Unerwartetes Token "s" in Ausdruck oder Anweisung.
WARNUNG: ConvertTo-DSCObject: "" (line 335): Die Instanzdefinition enthält einen unvollständigen
Eigenschaftszuweisungsblock.
WARNUNG: ConvertTo-DSCObject: "}" (line 784): Schließende ")" fehlt in einem Unterausdruck.
WARNUNG: ConvertTo-DSCObject: "s" (line 1871): Unerwartetes Token "s" in Ausdruck oder Anweisung.
WARNUNG: ConvertTo-DSCObject: "" (line 1871): Die Instanzdefinition enthält einen unvollständigen
Eigenschaftszuweisungsblock.
WARNUNG: ConvertTo-DSCObject: "" (line 1879): Nach "(" wurde ein Ausdruck erwartet.
```

#### Suggested solution to the issue
Check for apostrophys in strings at export an escape them, for example by doubling them ("'" -> "''".

#### The DSC configuration that is used to reproduce the issue (as detailed as possible)
See above

#### The operating system the target node is running
OsName               : Microsoft Windows 10 Enterprise
OsOperatingSystemSKU : EnterpriseEdition
OsArchitecture       : 64-Bit
WindowsVersion       : 2009
WindowsBuildLabEx    : 19041.1.amd64fre.vb_release.191206-1406
OsLanguage           : de-DE
OsMuiLanguages       : {de-DE}

#### Version of the DSC module that was used ('dev' if using current dev branch)
1.23.405.1, 1.23.412.1

**Comment by GruberMarkus** [comment]
This issue is very likely not limited to AADAdministrativeUnit, so other resources may have the same issue.

As the ps1 file created by the initial export is syntactically wrong, more features using the ps1 file may be affected, not only the conversion to an Excel file.

**Comment by GruberMarkus** [comment]
Hi @andikrueger and @NikCharlebois,

if there is anything I can do to help you solve this issue, just let me know.

**Comment by andikrueger** [comment]
Linking #3309 : This bug is most likely to be fixed within one of the DRG reused functions. There is a function to create these complex strings. It would be easy to combine this fix to escape the string properly within these functions and reuse this update across the codebase.

**Comment by GruberMarkus** [comment]
Hi @andikrueger and @NikCharlebois,

do you have any updates on this?

In release v1.23.1004.1, the export of AADAdministrativeUnit still creates invalid PowerShell code, as string characters are not escaped.

This makes it impossible to convert the export file into another format, no matter if JSON or XLSX.

An example (mind the unescaped apostrophe in the literal Identity string:
```
                MSFT_MicrosoftGraphMember{
                    Type = 'Device'
                    Identity = 'reini’s iPhone'
                }
```

If there is anything I can do to help you solve this issue, just let me know.

**Comment by GruberMarkus** [comment]
Solved in v1.23.1018.1. Thank you!
